src/sentry/web/frontend/oauth_token.py

```diff
@@ -128,7 +129,9 @@ def post(self, request: Request) -> HttpResponse:
     def get_access_tokens(self, request: Request, application: ApiApplication) -> dict:
         code = request.POST.get("code")
         try:
-            grant = ApiGrant.objects.get(application=application, code=code)
+            grant = ApiGrant.objects.get(
+                application=application, application__status=ApiApplicationStatus.active, code=code
```

Comment:
How can an application become inactive? 
Also this one has not been there before so I wonder if we need a clean up script to delete everything related to inactive apps?